src/stores/ProjectsInstitutionStore.js

```diff
@@ -48,22 +48,22 @@ export const useProjectInstitutionStore = defineStore({
       }
     },
 
-    async addInstitution(projectId, institutionId) {
+    async addInstitution(projectId, name) {
       const url = `${
         import.meta.env.VITE_API_URL
       }/projects/${projectId}/institutions/create`
 
-      const response = await axios.post(url, { institutionId })
+      const response = await axios.post(url, { name })
```

Comment:
If you want, you could also send `institution_id` or `name` (if it is a new institution, it wouldn't have an id and only send the id for pre-existing institutions). Then on the server, you can lookup using the primary key which is always faster the unique key.

Note, you can keep the logic that still lookup by 'name' since that is a safety check. But just add a lookup by primary key.